### PR TITLE
Make db_basic_test pass assert status checked

### PR DESCRIPTION
**Pull request by siying**
Summary:
Add db_basic_test status check list. Some of the warnings are suppressed. It is possible that some of them are due to real bugs.

Test Plan: See CI tests pass.

### CODE REVIEW

File: db/db_impl/db_impl_write.cc [hunk 1]
```diff
@@ -455,6 +456,7 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
   if (status.ok()) {
     status = w.FinalStatus();
   }
+  //  io_s.PermitUncheckedError();
```
Comment (zhichao-cao) [inline db/db_impl/db_impl_write.cc:459]: Should we remove it now?

Comment (siying) [reply db/db_impl/db_impl_write.cc:459]: Will do.

File: db/db_impl/db_impl_compaction_flush.cc [hunk 2]
```diff
@@ -246,16 +246,24 @@ Status DBImpl::FlushMemTableToOutputFile(
       // be pessimistic and try write to a new MANIFEST.
       // TODO: distinguish between MANIFEST write and CURRENT renaming
```
Comment (zhichao-cao) [inline db/db_impl/db_impl_compaction_flush.cc:247]: If it is convenient, you can also add  `.PermitUncheckedError()` to the remaining SetBGError at about L1193 and about L3000

https://github.com/facebook/rocksdb/pull/7342/files

**Review by zhichao-cao — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**